### PR TITLE
Editing pass

### PR DESCRIPTION
**Pull request by stevomatthews**
My initial set of edits to this document.

### CODE REVIEW

File: draft-sharabayko-mops-srt.md [hunk 1]
```diff
@@ -596,57 +679,100 @@ Last Acknowledged Packet Sequence Number (32 bits):
...
-The sender acknowledges (see ACKACK) only the receival of Full ACK packets.
+    - A Full ACK control packet is sent every 10 ms and has all the fields 
+      of {{ack-control-packet}}.
+    - A Lite ACK control packet includes only the Last Acknowledged Packet Sequence Number 
+      field. The Type-specific Information field should be set to 0.
+    - A Small ACK includes the fields up to and including the Available Buffer Size field. 
+      The Type-specific Information field should be set to 0.
 
-The Lite ACK and Small packets are used in cases when the receiver should acknowledge
+The sender only acknowledges only the receipt of Full ACK packets (see ACKACK).
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: "only" is used twice

File: draft-sharabayko-mops-srt.md [hunk 2]
```diff
@@ -656,10 +782,27 @@ Type-specific Information:
 Control Information Field:
 : This field must not appear in shutdown control packets.
 
+
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]:
```suggestion
```

File: draft-sharabayko-mops-srt.md [hunk 3]
```diff
@@ -671,17 +814,18 @@ Type-specific Information:
 Control Information Field:
 : This field must not appear in ACKACK control packets.
 
+
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]:
```suggestion
```

File: draft-sharabayko-mops-srt.md [hunk 4]
```diff
@@ -740,8 +888,17 @@ Too-Late Packet Drop ({{too-late-packet-drop}}) mechanisms are used in this mode
...
+portions of any size.
+
+
+## Stream Multiplexing
+
+Multiple SRT sockets may share one UDP socket, and the packets received on this
+UDP socket will be correctly dispatched to the SRT socket to which they are currently 
+destined. The parties exchange their SRT Socket IDs during the handshake. These IDs are 
+then used in the Destination Socket ID field of every control and data packet.
+
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: This paragraph was moved into the "SRT Data Transmission and Control" section above (line 831).
So this is a repetition.

File: draft-sharabayko-mops-srt.md [hunk 5]
```diff
@@ -804,33 +961,34 @@ Rendezvous handshake exchange has the following order of Handshake Types:
...
- | 1003 | REJ_RESOURCE     | Resource allocation problem |
- | 1004 | REJ_ROGUE        | incorrect data in handshake |
- | 1005 | REJ_BACKLOG      | listener's backlog exceeded |
- | 1006 | REJ_IPE          | internal program error |
- | 1007 | REJ_CLOSE        | socket is closing |
+When a connection process has failed before either party can send the CONCLUSION handshake, 
+the Handshake Type field will contain the appropriate error value for the rejected 
+connection. See the list of error codes in {{hs-rej-reason}}.
+
+ | Code | Error            | Description                                    |
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: Maybe "Error code" is a better name then?

File: draft-sharabayko-mops-srt.md [hunk 6]
```diff
@@ -1156,58 +1315,61 @@ never become aware. The missing packet problem is resolved this way:
...
+which is decided by adding the timestamp to the configured latency).
+
 It is worth mentioning that the use of the packet sending time to stamp the packets is
-inappropriate for TSBPD feature since a new time (current sending time) is used for retransmitted packets,
-putting them out of order when inserted at their proper place in the stream. Packets are
-inserted based on the sequence number in the header field. The origin time (in microseconds)
-of the packet is already sampled when a packet is first submitted by the application to the SRT sender.
-The TSBPD feature uses this time to stamp the packet for first transmission and any subsequent retransmission.
-This timestamp and the configured SRT latency ({{srt-latency}}) control the recovery buffer size and the instant that packets
-are delivered at the destination.
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: This part starting from "Packets are inserted .." is deleted. Was it an intent?

File: draft-sharabayko-mops-srt.md [hunk 7]
```diff
@@ -1506,19 +1667,32 @@ In addition to the regular NAKs, the Periodic NAK report mechanism can be used t
...
+are sent directly and processed upon reception, but ACKACK processing time is negligible 
+(the time this takes is included in the round-trip time).
+
 
 
 ## Bidirectional Transmission Queues
 
 Once an SRT connection is established, both peers can send data packets simultaneously.
 
+
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]:
```suggestion
```

File: draft-sharabayko-mops-srt.md [hunk 8]
```diff
@@ -1506,19 +1667,32 @@ In addition to the regular NAKs, the Periodic NAK report mechanism can be used t
...
 ## Round Trip Time Estimation
 
 The round-trip time is estimated during the transmission of SRT data packets
 based on the time difference between the ACK packet is sent and the
 corresponding ACKACK is received by the data receiver.
 
 
+
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]:
```suggestion
```

File: draft-sharabayko-mops-srt.md [hunk 9]
```diff
@@ -596,57 +679,100 @@ Last Acknowledged Packet Sequence Number (32 bits):
...
 
+An SRT SHUTDOWN Control packet is formatted as follows:
+
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ---
+|1|        Control Type         |           Reserved            |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                   Type-specific Information                   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ SRT Header
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: This line is 77 characters in width. The conversion to IETF txt format requires 72.
Maybe we can fix it this way:
```
+-+-+-+-+-+-+-+-+SRT Header-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ 
```

File: draft-sharabayko-mops-srt.md [hunk 10]
```diff
@@ -596,57 +679,100 @@ Last Acknowledged Packet Sequence Number (32 bits):
...
+|                           Time Stamp                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                     Destination Socket ID                     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+ ---
+|                              None                             | CIF
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: CIF = None to reduce the width?

File: draft-sharabayko-mops-srt.md [hunk 11]
```diff
@@ -701,36 +845,40 @@ In Live Transmission Mode the only valid value is "1".
...
+
 
 ### Live Mode {#transmission-mode-live}
 
-Live mode is a special case of the message mode where only data packets
-with PP field set to "11b" are allowed.
+Live mode is a special type of message mode where only data packets
+with their PP field set to "11b" are allowed.
+
+Additionally, TsbPd ({{tsbpd}}) and TL Packet drop ({{tl-pkt-drop}}) mechanisms are used
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: Section anchor has been changed.
```suggestion
Additionally, TsbPd ({{tsbpd}}) and TL Packet drop ({{too-late-packet-drop}}) mechanisms are used 
```

File: draft-sharabayko-mops-srt.md [hunk 12]
```diff
@@ -701,36 +845,40 @@ In Live Transmission Mode the only valid value is "1".
...
-it is only allowed for an application to send and receive the whole message.
-Until the message is complete (all packets are received) it will not be allowed to read it.
-
-The Order Flag of the Data packet set to 1 restricts the reading order of the messages to be sequential.
-While the Order Flag set to 0 allows to read those messages that are already fully available, before
-preceding messages, that still have some packets missing.
+set to 1. Thus, a PP equal to "11b" indicates a packet that forms the whole message.
+A PP equal to "00b" indicates a packet that belongs to the inner part of the message.
+
+The concept of the message in SRT comes from UDT ({{I-D.gg-udt}}). In this mode a single
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]:
```suggestion
The concept of the message in SRT comes from UDT ({{GHG04b}}). In this mode a single 
```

File: draft-sharabayko-mops-srt.md [hunk 13]
```diff
@@ -351,23 +355,24 @@ Handshake Type (32 bits):
  | 0xFFFFFFFD | DONE                         |
  | 0xFFFFFFFE | AGREEMENT                    |
  | 0xFFFFFFFF | CONCLUSION                   |
- | 0x00000000 | WAVEHAND                     |
+ | 0x00000000 | WAVEAHAND                    |
```
[the remark this answers is not in the document]
Comment (stevomatthews) [reply draft-sharabayko-mops-srt.md]: This is how it is referred to in Mikolaj's handshake document.

File: draft-sharabayko-mops-srt.md (hunk 1 above)
Comment (stevomatthews) [reply draft-sharabayko-mops-srt.md]: Good catch!

File: draft-sharabayko-mops-srt.md (hunk 4 above)
Comment (stevomatthews) [reply draft-sharabayko-mops-srt.md]: Will delete it.

File: draft-sharabayko-mops-srt.md (hunk 6 above)
Comment (stevomatthews) [reply draft-sharabayko-mops-srt.md]: Will replace

File: draft-sharabayko-mops-srt.md [hunk 14]
```diff
@@ -471,69 +473,93 @@ The value of request is 3, and the response value is 4.
...
+- 7: Reserved to discriminate MPEG-TS packet (0x47=sync byte)   
 
-KK (2 bits):
-: Encryption Flag. Refer to KK field in {{data-pkt}}.
 
-KEKI (32 bits):
-: Key Encryption Key Index in big endian order. The default stream associated
-  key set to 0. The values from 1 to 255 are reserved for manually indexed keys.
+Signature (Sign): 16 bits. Value: {0x2029}  
+: This is a fixed-width field that contains the signature ‘HAI‘ encoded as a PnP Vendor ID
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: PNPID reference was here

File: draft-sharabayko-mops-srt.md [hunk 15]
```diff
@@ -471,69 +473,93 @@ The value of request is 3, and the response value is 4.
...
-: Encryption Cipher and mode. 
+Reserved (Resv): 6 bits. Value: {0}  
+: This is a fixed-width field reserved for flag extension or other usage.
 
- | Bitmask    | Flag |
- | ---------- | :---------------: |
- | 0x00       | None or KEK indexed crypto context |
- | 0x01       | AES-ECB (potentially for VF 2.0 compatible message) |
- | 0x02       | AES-CTR ({{SP800-38A}}) |
-{: #hs-km-msg-flags title="HS Key Material Exchange Cipher mode"}
```
Comment (maxsharabayko) [inline draft-sharabayko-mops-srt.md]: SP800-38A reference is missing now

**Pull request merged**